tests/framework/utils_cpuid.py

```diff
@@ -63,12 +63,12 @@ def get_cpu_codename():
     vendor = get_cpu_vendor()
     if vendor == CpuVendor.INTEL:
         result = re.match(r"^(.*) @.*$", cpu_model)
-        assert result, "CPU model name does not match the regex pattern."
-        return CPU_DICT[CpuVendor.INTEL].get(result.group(1), "Unknown")
+        if result:
+            return CPU_DICT[CpuVendor.INTEL].get(result.group(1), "Unknown")
     if vendor == CpuVendor.AMD:
         result = re.match(r"^(.*) [0-9]*-Core Processor$", cpu_model)
-        assert result, "CPU model name does not match the regex pattern."
-        return CPU_DICT[CpuVendor.AMD].get(result.group(1), "Unknown")
+        if result:
+            return CPU_DICT[CpuVendor.AMD].get(result.group(1), "Unknown")
     if vendor == CpuVendor.ARM:
         return cpu_model
     return None
```

Comment:
Not directly related to this PR, but how about changing to return "Unknown" for unknown CPU vendor?

Reply:
+1

Reply:
+1 for consistency. How about `None` for all cases, or better yet, follow the Python convention of letting the caller choose the default:

```python
def get_cpu_codename(default=None):
    ...
    return default
```

Reply:
Adding some background on the "Unknown" : https://github.com/firecracker-microvm/firecracker/pull/3627#discussion_r1175352075

Reply:
I have updated to return a default of `"Unknown"`.